WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/WebView.swift

```diff
@@ -72,6 +72,10 @@ struct WebView: UIViewRepresentable {
         if shouldReloadOnUpdate {
             uiView.reload()
         }
+
+        if uiView.url != url {
+            uiView.load(URLRequest(url: url))
+        }
```

Comment:
Can you explain this code? 

1. Would this affect other screens outside the Themes Preview screen? (As this isn't checking for `shouldReloadOnUpdate` value)
1. How are `reload` and `load` different? Do we have to do both function calls?

Reply:
1. `shouldReloadOnUpdate` is used for the layout switcher's functionality. Layout switching depends on re-applying Javascript on the same URL, and the re-applying is done by reloading that same URL.
2. The `uiView.load()` call is used for the page switcher functionality. To load a new page, the parent View passes the new URL to this `WebView`, and `updateUIView()` gets called to ask the actual `WKWebView` to load that new page.

Reply:
I added a comment in https://github.com/woocommerce/woocommerce-ios/pull/11492/commits/297d47791683669dcdb05e6234bda10fdc670b87 to explain point 2 above.

Reply:
> Can you explain what you meant with this?

- `uiView.load(URLRequest(url: url))` will execute even if `shouldReloadOnUpdate` is false. 
- But, if we take `uiView.reload()` method call, checking `shouldReloadOnUpdate` before calling the method ensures that this change won't affect other screens outside Themes. (As currently `shouldReloadOnUpdate` value is set as `true` only when used in `ThemesPreviewView`)

The `uiView.load(URLRequest(url: url))` will be executed for all other screens where `WebView` is used. We need to ensure that this change won't have any undesired effects. 

What do you think? 

1. Should the `uiView.load(URLRequest(url: url))` call be executed for all screens using `WebView`? 
1. What happens when we call `uiView.reload()` first and then `uiView.load(URLRequest(url: url))` for the themes screen case? Is it necessary to call both methods? 

Tagging @itsmeichigo to get their opinion on this.

Reply:
> Should the uiView.load(URLRequest(url: url)) call be executed for all screens using WebView?

Yes, if the parent View in other screen needs to programmatically update the URL in the `WebView` call (like what's happening here when changing Page manually), they want this to be called so that the new URL is actually loaded.

I wonder if `shouldReloadOnUpdate` can be renamed to something else to make it clearer what it does. Here is the logic with the layout change:

1. When a new layout is selected in the menu, then `selectedDevice` is updated,
2. This in turns redraws the `WebView` since the `onCommit` parameter is updated:
```
                    WebView(
                        isPresented: .constant(true),
                        url: url,
                        shouldReloadOnUpdate: true,
                        disableLinkClicking: true,
                        onCommit: { webView in
                            webView.evaluateJavaScript(self.selectedDevice.viewportScript)
                        }
                    )
```
3. For the new JavaScript in `onCommit` to be called, the page also needs to be reloaded, since the JS call depends on the `didCommit` event below to happen.

```
        func webView(_ webView: WKWebView, didCommit navigation: WKNavigation!) {
            parent.onCommit?(webView)
        }
```
4. By default this is not happening, though, since the `url` passed in the `WebView` call remains the same, thus the webview won't reload it.
5. Thus, we need to also force the reload somehow. This is where the `shouldReloadOnUpdate` parameter comes in. If this is true, then whenever there's any `WebView` redraw/update, then it should reload the screen. This in turn calls the JavaScript again, which is needed to change the layout.

> What happens when we call uiView.reload() first and then uiView.load(URLRequest(url: url)) for the themes screen case? Is it necessary to call both methods?

I think I see what you mean, though. I suppose it should be like this instead?
```
        if shouldReloadOnUpdate && uiView.url == url {
            uiView.reload()
        }
```

Reply:
I think it makes sense to load the request on state change. I found [an article](https://sarunw.com/posts/swiftui-webview/) that advises loading requests even without checking for URL changes. I experimented with this and found that reloading is not necessary if we do that:

```Swift
    func updateUIView(_ uiView: WKWebView, context: Context) {
        uiView.load(URLRequest(url: url))
    }
```

The reason is that the web view is rendered again when the preview view's `selectedDevice` changes, which triggers `updateUIView`. The web view is then asked to load the given URL again and the correct layout is rendered with the injected JS script. If we follow this, the load request `makeUIView` is redundant.

Reply:
Nice. Thanks Huong! 🙇 

@hafizrahman  Let us follow Huong's suggestion and get rid of `shouldReloadOnUpdate`.

Reply:
Nice info in that article. Makes me wonder why `fun updateUIView()` was empty before, perhaps because the `WebView` is not expected to handle state change. I'll update & simplify the code.

Reply:
Added in https://github.com/woocommerce/woocommerce-ios/pull/11492/commits/91c59aa7db17b438bab2a5357d871977bd1e027e